docs/developer/architecture/core/saved-objects-service.asciidoc

```diff
@@ -95,48 +97,50 @@ Each Saved Object type can define it's own {es} field mappings.
 Because multiple Saved Object types can share the same index, mappings defined
 by a type will be nested under a top-level field that matches the type name.
 
-For example, the mappings defined by the `dashboard_visualization` Saved
+For example, the mappings defined by the `search` Saved
 Object type:
 
-.src/plugins/my_plugin/server/saved_objects/dashboard_visualization.ts
+src/plugins/saved_search/server/saved_objects/search.ts
 [source,typescript]
 ----
 import { SavedObjectsType } from 'src/core/server';
-
-export const dashboardVisualization: SavedObjectsType = {
-  name: 'dashboard_visualization',
-  ...
-  mappings: {
-    properties: {
+// ... other imports
+export function getSavedSearchObjectType(
+  getSearchSourceMigrations: () => MigrateFunctionsObject
+): SavedObjectsType {
+  return {
```

Comment:
Nit: imo our examples can simplify reality a bit by avoiding `MigrateFunctionsObject` in favour of `SavedObjectsType`. No need for everyone to wrap their SO definitions in a function :D 

```suggestion
  getSearchSourceMigrations:  SavedObjectsType = {
```

Reply:
This specific example is directly from the `search` SO definition and I've linked to it above the code snippet.
If we change the definition, which I think should be fine, then I'll have to remove the link.

Reply:
I've kept the link, made the change to a simplified version and commented as such below the code snippet.